specfile/changelog.py

```diff
@@ -0,0 +1,127 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import collections
+import datetime
+import re
+from typing import List
+
+from specfile.sections import Section
+
+DAYS = [
+    "Mon",
+    "Tue",
+    "Wed",
+    "Thu",
+    "Fri",
+    "Sat",
+    "Sun",
+]
+
+MONTHS = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+]
+
+
+class ChangelogEntry:
+    """
+     Class that represents a changelog entry.
+
+    Attributes:
+        date: Date of the entry.
+        author: Author of the entry.
+        evr: EVR (epoch, version, release) of the entry.
+        content: List of lines forming the content of the entry.
+    """
+
+    def __init__(
+        self, date: datetime.date, author: str, evr: str, content: List[str]
+    ) -> None:
+        self.date = date
+        self.author = author
+        self.evr = evr
+        self.content = content.copy()
+
+    def __str__(self) -> str:
+        return f"{self.header}\n" + "\n".join(self.content) + "\n"
+
+    def __repr__(self) -> str:
+        date = repr(self.date)
+        content = repr(self.content)
+        return f"ChangelogEntry({date}, '{self.author}', '{self.evr}', {content})"
+
+    @property
+    def header(self) -> str:
+        """Entry header in the format that is being used in Fedora spec files."""
+        weekday = DAYS[self.date.weekday()]
+        month = MONTHS[self.date.month - 1]
+        return f"* {weekday} {month} {self.date:%d %Y} {self.author} - {self.evr}"
+
+
+class Changelog(collections.UserList):
+    """
+    Class that represents a changelog.
+
+    Attributes:
+        data: List of individual entries.
+    """
+
+    def __str__(self) -> str:
+        return "\n".join(str(i) for i in self.data)
+
+    def __repr__(self) -> str:
+        data = repr(self.data)
+        return f"Changelog({data})"
+
+    @staticmethod
+    def parse(section: Section) -> "Changelog":
+        """
+        Parses a %changelog section.
+
+        Args:
+            section: Section to parse.
+
+        Returns:
+            Constructed instance of `Changelog` class.
+        """
+        header_regex = re.compile(r"^\* \w+ (\w+) (\d{2}) (\d{4}) (.+) - (.+)$")
+        data = []
+        for line in section:
+            m = header_regex.match(line)
+            if m:
```

Comment:
Can we use [walrus](https://www.python.org/dev/peps/pep-0572/) here? (What Python versions do we want to support?)

```suggestion
            if m := header_regex.match(line):
```

Reply:
I think we should support the same Python versions as packit. Are there plans to drop support for Python 3.6 and 3.7?

But nice idea, I didn't even think of it.

Reply:
Would be nice to use it but I recently used something specific from python 3.8 (where walrus operator was also added) and the CI was failing in EPEL.

Reply:
Good point with EPEL.